File: ios/NativeModules/CustomPlayerViewManager.m
#import <Foundation/Foundation.h>
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(CustomPlayerViewManager, RCTViewManager)
  RCT_EXPORT_VIEW_PROPERTY(autoPlay, NSString)
  RCT_EXPORT_VIEW_PROPERTY(zoom, NSNumber)
@end

File: ios/NativeModules/CustomPlayerViewManager.swift
import Foundation
import UIKit
import PlayerKit
import AVFoundation

@objc(CustomPlayerViewManager)
class CustomPlayerViewManager: RCTViewManager {

  
  override func view() -> UIView! {
    return PlayerView(frame: CGRect(x:0, y:0, width: 300, height: 300))
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
